docs/design/features/Apple-Silicon-UX.md

```diff
@@ -0,0 +1,107 @@
+
+- The installer will be modified to enforce the .NET 6 architecture policy. It will prevent osx-x64 install on Apple Silicon
+
+- The existing installer releases will need to be tested to make sure they do not clobber the new `dotnet` modifications.
+
+#### Implications
+
+- The proposed version policy will prohibit us from making a native port of earlier runtime releases. .NET 2.1 will be out of support before .NET 6.0 releases. .NET 5.0 will be out of support 3 month after release of .NET 6. .NET 3.1 is the only viable candidate for back porting Apple Silicon support.  However, given the size of the changes to the runtime and the JIT it is highly unlikely this is something we would actively pursue.
+
+- The proposed version policy will prevent .NET 6 apps published with rid of osx-x64 from running on Apple Silicon even with Rosetta 2 emulation. This is desirable because long term we want to stop supporting Rosetta 2 emulation on Apple Silicon. IT support grows our effective platform support matrix and increases our support burden. We are dependent on Apple to fix Rosetta 2 issues, but they are dependent on us to root cause them. Performance is also expected to be better with native code.
+
+#### Future optimizations
+
+The current proposed design will perform the runtime search twice when an architecture mismatch is detected. The user can prevent this extra delay by calling dotnet with the correct architecture for the required runtime.
+
+- The default architecture is selected based on the architecture of the current process (typically the shell or the IDE).
+- The user or the IDE can explicitly select the architecture of the target process using the `arch` command.
+
+The extra search cost could be minimized by passing the result of the previous search when restarting the process. Adding an `--fx-version` option would accomplish this at the cost of changing the users specified argument order.  Using a new environment variable would avoid this unexpected side-effect.
+
+## Apple Silicon requires Signed Binaries
+
+Apple Silicon with Big Sur adds a new security policy which requires signed binaries by default. Unsigned binaries are simply killed without explanation. The user can expect an inscrutable message like `zsh killed: helloWorld` This directly affects our users who expect published binaries to just work.
+
+The developer would need to explicitly sign after publishing for Apple Silicon.  The sigining command is relatively simple `codesign -s <signature> <app-path>`.
```

Comment:
Are you saying that `dotnet run` will always fail on Apple Silicon?

Reply:
`dotnet run` will work because the release version of `dotnet` is signed by Microsoft.  

The published app will be killed until it is signed.
```console
dotnet publish -r osx-arm64
helloWorld -> bin/Debug/net6.0/osx-arm64/publish

# helloWorld AppHost needs codesign -s <signature> bin/Debug/net6.0/osx-arm64/publish/helloWorld

bin/Debug/net6.0/osx-arm64/publish/helloWorld
zsh killed: helloWorld

# Sign it
codesign -s <signature> bin/Debug/net6.0/osx-arm64/publish/helloWorld

# Now it works...
bin/Debug/net6.0/osx-arm64/publish/helloWorld
Hello World
```

...publish
```

Reply:
To deal with the signing issue, `dotnet run` has been special cased on macOS to start the app using `dotnet`, e.g. `dotnet helloWorld.dll`. 

On all other platforms, `dotnet run` will just start the app using apphost, e.g. `helloWorld` since it provides better experience, like you can see the actual app name in the process list and not just generic `dotnet`.

Reply:
https://docs.microsoft.com/en-us/dotnet/core/install/macos-notarization-issues#apphost-is-disabled-by-default